### PR TITLE
fix(Scripts/IcecrownCitadel): fix Lord Marrowgar bone spikes

### PR DESCRIPTION
**Pull request by Nyeriah**
## Changes Proposed:
-  
-  

## Issues Addressed:

- Closes 

## Tests Performed:

- 
- 

## How to Test the Changes:

1.
2.
3.

## Known Issues and TODO List:

- [ ]
- [ ]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

File: src/server/scripts/Northrend/IcecrownCitadel/boss_lord_marrowgar.cpp [hunk 1]
```diff
@@ -408,27 +408,31 @@ class npc_bone_spike : public CreatureScript
         {
             if (!summoner)
                 return;
-
-            if (summoner->GetTypeId() != TYPEID_UNIT)
+
```
Comment (Nyeriah) [inline src/server/scripts/Northrend/IcecrownCitadel/boss_lord_marrowgar.cpp:411]:
```suggestion

```

File: src/server/scripts/Northrend/IcecrownCitadel/boss_lord_marrowgar.cpp [hunk 2]
```diff
@@ -408,27 +408,30 @@ class npc_bone_spike : public CreatureScript
         {
             if (!summoner)
                 return;
-
-            if (summoner->GetTypeId() != TYPEID_UNIT)
+            if (Unit* summonerUnit = summoner->ToUnit())
             {
-                return;
+                if (Vehicle* v = summoner->ToUnit()->GetVehicle())
```
Comment (Nyeriah) [inline src/server/scripts/Northrend/IcecrownCitadel/boss_lord_marrowgar.cpp:413]:
```suggestion
                if (Vehicle* v = summonerUnit->GetVehicle())
```

File: src/server/scripts/Northrend/IcecrownCitadel/boss_lord_marrowgar.cpp [hunk 3]
```diff
@@ -408,27 +408,30 @@ class npc_bone_spike : public CreatureScript
...
+                    if (Unit* u = v->GetBase())
+                    {
+                        if (u->GetEntry() == NPC_BONE_SPIKE && u->GetTypeId() == TYPEID_UNIT)
+                        {
+                            u->ToCreature()->AI()->DoAction(-1337);
+                        }
+                    }
+                }
+                
+                ObjectGuid petGUID = summoner->ToUnit()->GetPetGUID();
```
Comment (Nyeriah) [inline src/server/scripts/Northrend/IcecrownCitadel/boss_lord_marrowgar.cpp:424]:
```suggestion
                ObjectGuid petGUID = summonerUnit->GetPetGUID();
```

File: src/server/scripts/Northrend/IcecrownCitadel/boss_lord_marrowgar.cpp [hunk 4]
```diff
@@ -408,27 +408,30 @@ class npc_bone_spike : public CreatureScript
...
+                        if (u->GetEntry() == NPC_BONE_SPIKE && u->GetTypeId() == TYPEID_UNIT)
+                        {
+                            u->ToCreature()->AI()->DoAction(-1337);
+                        }
+                    }
+                }
+                
+                ObjectGuid petGUID = summonerUnit->GetPetGUID();
+                summonerUnit->SetPetGUID(ObjectGuid::Empty);
+                me->CastSpell(summoner->ToUnit(), SPELL_IMPALED, true);
```
Comment (Nyeriah) [inline src/server/scripts/Northrend/IcecrownCitadel/boss_lord_marrowgar.cpp:426]:
```suggestion
                me->CastSpell(summonerUnit, SPELL_IMPALED, true);
```

File: src/server/scripts/Northrend/IcecrownCitadel/boss_lord_marrowgar.cpp [hunk 5]
```diff
@@ -408,27 +408,30 @@ class npc_bone_spike : public CreatureScript
...
+                {
+                    if (Unit* u = v->GetBase())
+                    {
+                        if (u->GetEntry() == NPC_BONE_SPIKE && u->GetTypeId() == TYPEID_UNIT)
+                        {
+                            u->ToCreature()->AI()->DoAction(-1337);
+                        }
+                    }
+                }
+
```
Comment (Nyeriah) [inline src/server/scripts/Northrend/IcecrownCitadel/boss_lord_marrowgar.cpp:423]:
```suggestion

```

**Review by Gultask — APPROVED** [review]

**Comment by Tony931023** [comment]
works perfect, thanks @Nyeriah

**Pull request merged**